cheroot/workers/threadpool.py

```diff
@@ -173,17 +173,7 @@ def __init__(
 
     def start(self):
         """Start the pool of threads."""
-        for _ in range(self.min):
-            self._threads.append(WorkerThread(self.server))
-        for worker in self._threads:
-            worker.name = (
-                'CP Server {worker_name!s}'.
-                format(worker_name=worker.name)
-            )
-            worker.start()
-        for worker in self._threads:
-            while not worker.ready:
-                time.sleep(.1)
+        self.grow(self.min)
```

Comment:
`grow()`'s logic doesn't always spawn the same amount of threads as the current implementation.
With the default `max = -1`, it calculates the same amount of threads as the current logic. But if one were to set like `max = 2`, then the `budget` would be calculated as `2` and not `min = 10` (the current default).

So the difference is that the current implementation spawns exactly the amount of threads requested by the `min` argument, while this PR's implementation may spawn less in case of `max < min`.

Reply:
I guess the real question is what should the code do if the user runs something like `ThreadPool(max=2)`?
 - Spawn 10 threads (current)
 - Spawn 2 threads (this PR)
 - `raise ValueError("min of 10 cannot be greater than max of 2")`

Reply:
That's an interesting observation too. Usually, in such cases I go for Git archeology (inspecting what added this behavior / commits / blame / motivation). Do you think you could look into it and attempt to deduce what the original motivation was?

Reply:
Following up on "which behavior of double-start is more correct", I think it could follow stdlib's Thread behavior and throw a runtime error:
```python-traceback
$ python 
Python 3.11.0rc2 (main, Sep 24 2022, 17:37:14) [GCC 11.2.1 20220115] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from threading import Thread
>>> thread = Thread()
>>> thread.start()
>>> thread.start()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/.pyenv/versions/3.11.0rc2/lib/python3.11/threading.py", line 952, in start
    raise RuntimeError("threads can only be started once")
RuntimeError: threads can only be started once
```

And re:input validation — it sounds like throwing an exception is the right thing to do here as well. But in an attempt to avoid https://xkcd.com/1172/ I'd like to understand how this was being designed originally.

Reply:
Did some digging:

The original implementation of the class in e8127b98 would not raise any errors when calling `start()` twice and just add `self.min` more threads each time. `.grow()` would refuse to add more than `.max` threads. `.start()` would add all the threads to `._threads`, start them all, then wait for them all to be ready. `.grow()` would create the new thread, add it to `._threads`, then start it without waiting for it to be ready.

47844dab changed `.grow()` to spawn all the workers before adding them to `._threads`, still not waiting for them to be ready. I couldn't figure out why this was changed.

1d70c4b1 changed `.grow()` to wait for threads to become ready before adding them to `._threads`. The commit message mentioned it fixed issue #1205 which seems to be a reference to an old bug tracker.